src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1382,12 +1386,6 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 {
                     AddSynthesizedAttribute(ref attributes, compilation.SynthesizeTupleNamesAttribute(baseType));
                 }
-
-                if (compilation.ShouldEmitNullableAttributes(this) &&
```

Comment:
>if (compilation.ShouldEmitNullableAttributes(this) && [](start = 16, length = 53)

It is not clear why this was moved to the base, but the dynamic and tuple element name handling wasn't.